src/core/server/saved_objects/service/lib/filter_utils.ts

```diff
@@ -28,11 +28,12 @@ const astFunctionType = ['is', 'range', 'nested'];
 
 export const validateConvertFilterToKueryNode = (
   allowedTypes: string[],
-  filter: string,
+  filter: string | KueryNode,
   indexMapping: IndexMapping
 ): KueryNode | undefined => {
-  if (filter && filter.length > 0 && indexMapping) {
-    const filterKueryNode = esKuery.fromKueryExpression(filter);
+  if (filter && indexMapping) {
+    const filterKueryNode =
+      typeof filter === 'string' ? esKuery.fromKueryExpression(filter) : filter;
```

Comment:
NIT: we are no longer ignoring empty string filters, but I guess this has no impact?

Reply:
Empty strings are still ignored, I added https://github.com/elastic/kibana/pull/75693/commits/195a5a166c0164187142470f7ed9b8e249d28553#diff-c5ab35755205b8adbf6cfc858d69ea3bR86-R88 to double-check this. JavaScript treats an empty-string as being falsy, so checking `filter` and `filter.length > 0` was redundant.

Reply:
Wait what? Kibana is not Java code?

Reply:
Unfortunately, no... Maybe in the future though! New-new platform??

Reply:
We would need to find a more suitable codename if we switch to java. NewPlatformBijectiveAdapterFactory maybe.